adapters/seedingAlliance/seedingAlliance.go

```diff
@@ -0,0 +1,142 @@
+	for i := range request.Imp {
+		if err := addTagID(&request.Imp[i]); err != nil {
+			return nil, []error{err}
+		}
+	}
+
+	request.Cur = append(request.Cur, "EUR")
+
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestData := &adapters.RequestData{
+		Method: http.MethodPost,
+		Uri:    a.endpoint,
+		Body:   requestJSON,
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *SeedingAllianceAdapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+
+	for _, seatBid := range response.SeatBid {
+		for i := range seatBid.Bid {
+			resolvePriceMacro(&seatBid.Bid[i])
+
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: getMediaTypeForImp(seatBid.Bid[i].ImpID, request.Imp),
+			}
+
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+
+	return bidResponse, nil
+}
+
+func resolvePriceMacro(bid *openrtb2.Bid) {
+	price := strconv.FormatFloat(bid.Price, 'f', -1, 64)
+	bid.AdM = strings.Replace(bid.AdM, "${AUCTION_PRICE}", price, -1)
+}
+
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	var UnsupportedBidType openrtb_ext.BidType = "unsupported"
+	var UnknownBidType openrtb_ext.BidType = "unknown"
+
+	for _, imp := range imps {
+		if imp.ID == impId {
+			switch {
+			case imp.Native != nil:
+				return openrtb_ext.BidTypeNative
+			case imp.Banner != nil:
+				return openrtb_ext.BidTypeBanner
+			case imp.Video != nil:
+				return UnsupportedBidType
+			case imp.Audio != nil:
+				return UnsupportedBidType
+			default:
+				return UnknownBidType
+			}
+		}
+	}
+
+	return UnknownBidType
+}
+
+func addTagID(imp *openrtb2.Imp) error {
+	var ext adapters.ExtImpBidder
+	var extSA openrtb_ext.ImpExtSeedingAlliance
+
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return fmt.Errorf("could not unmarshal adapters.ExtImpBidder: %w", err)
+	}
+
+	if err := json.Unmarshal(ext.Bidder, &extSA); err != nil {
+		if v, ok := ext.Prebid.Bidder["seedingAlliance"]; ok {
```

Comment:
There is no need for this check. Prebid Server gives it to adapter in the same way regardless if imp.ext looks like this 
```
"ext": {
                "seedingAlliance": {
                    "adunitid": "test1"
                }
            }
```
or like this:
```
"ext": {
                "prebid": {
                    "bidder": {
                        "seedingAlliance": {
                            "adunitid": "test1"
                        }
                    }
                }
            }
```